package/MDAnalysis/coordinates/timestep.pyx

```diff
@@ -0,0 +1,982 @@
+        if has_velocities:
+            ts.velocities = velocities
+        if has_forces:
+            ts.forces = forces
+
+        return ts
+
+    def __eq__(self, other):
+        """Compare with another Timestep
+
+        .. versionadded:: 0.11.0
+        """
+        if not isinstance(other, Timestep):
+            return False
+
+        if not self.frame == other.frame:
+            return False
+
+        if not self.n_atoms == other.n_atoms:
+            return False
+
+        if not self.has_positions == other.has_positions:
+            return False
+        if self.has_positions:
+            if not (self.positions == other.positions).all():
+                return False
+
+        if self.dimensions is None:
+            if other.dimensions is not None:
+                return False
+        else:
+            if other.dimensions is None:
+                return False
+            if not (self.dimensions == other.dimensions).all():
+                return False
+
+        if not self.has_velocities == other.has_velocities:
+            return False
+        if self.has_velocities:
+            if not (self.velocities == other.velocities).all():
+                return False
+
+        if not self.has_forces == other.has_forces:
+            return False
+        if self.has_forces:
+            if not (self.forces == other.forces).all():
+                return False
+
+        return True
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __getitem__(self, atoms):
+        """Get a selection of coordinates
+
+        ``ts[i]``
+
+           return coordinates for the i'th atom (0-based)
+
+        ``ts[start:stop:skip]``
+
+           return an array of coordinates, where start, stop and skip
+           correspond to atom indices,
+           :attr:`MDAnalysis.core.groups.Atom.index` (0-based)
+        """
+        if isinstance(atoms, numbers.Integral):
+            return self._pos[atoms]
+        elif isinstance(atoms, (slice, np.ndarray)):
+            return self._pos[atoms]
+        else:
+            raise TypeError
+
+    def __getattr__(self, attr):
+        # special-case timestep info
+        if attr in ('velocities', 'forces', 'positions'):
+            raise NoDataError('This Timestep has no ' + attr)
+        err = "{selfcls} object has no attribute '{attr}'"
+        raise AttributeError(err.format(selfcls=type(self).__name__,
+                                        attr=attr))
+
+    def __len__(self):
+        return self.n_atoms
+
+    def __iter__(self):
+        """Iterate over coordinates
+
+        ``for x in ts``
+
+            iterate of the coordinates, atom by atom
+        """
+        for i in range(self.n_atoms):
+            yield self[i]
+
+    def __repr__(self):
+        desc = "< Timestep {0}".format(self.frame)
+        try:
+            tail = " with unit cell dimensions {0} >".format(self.dimensions)
+        except NotImplementedError:
+            tail = " >"
```

Comment:
This might not be being caught because `.dimensions` is always present but sometimes `None`

Reply:
I don't think its now possible or sensible to catch anything here anyway so I will remove the `try: except: block`